--- a/accelerator/CMakeLists.txt   2026-04-12 14:03:40.942659316 +0200
+++ b/accelerator/CMakeLists.txt   2026-04-12 14:05:57.374512004 +0200
@@ -42,7 +42,7 @@
 casparcg_add_library(accelerator SOURCES ${SOURCES} ${HEADERS})
 target_include_directories(accelerator PRIVATE .. ${CMAKE_CURRENT_BINARY_DIR})
 target_precompile_headers(accelerator PRIVATE StdAfx.h)
-target_link_libraries(accelerator PRIVATE common core GLEW::glew)
+target_link_libraries(accelerator PRIVATE common core GLEW::glew OpenGL::EGL)
 target_link_libraries(accelerator PRIVATE sfml-window)
 
 source_group(sources ./.*)
